tensorflow/compiler/tf2tensorrt/convert/ops/variable_ops.cc

```diff
@@ -217,6 +217,194 @@ class ConvertVariableV2 : public OpConverterBase<ConvertVariableV2> {
 REGISTER_DEFAULT_TRT_OP_CONVERTER(MakeConverterFunction<ConvertVariableV2>(),
                                   {"VariableV2"});
 
+class ConvertReadVariableOp : public OpConverterBase<ConvertReadVariableOp> {
+ public:
+  ConvertReadVariableOp(OpConverterParams* params)
+      : OpConverterBase<ConvertReadVariableOp>(params) {}
+
+  struct ReadVariableOpAttributes {
+    TensorShapeProto shape_proto;
+    TensorShape shape;
+    string name;
+    DataType dtype;
+  };
+
+  static constexpr std::array<InputArgSpec, 1> InputSpec() {
+    return {InputArgSpec::Create("resource", TrtInputArg::kResource)};
+  }
+
+  static constexpr std::array<DataType, 2> AllowedDataTypes() {
+    return {DataType::DT_FLOAT, DataType::DT_HALF};
+  }
+
+  static constexpr const char* NodeDefDataTypeAttributeName() {
+    return "dtype";
+  }
+
+  template <typename T>
+  Status ValidateImpl() {
+    const auto& node_def = params_->node_def;
+
+    // Verify and consume node attributes.
+    StatusOr<TensorShapeProto> shape_proto =
+        GetAttrValue<TensorShapeProto>("_shape");
+    TRT_ENSURE_OK(shape_proto);
+
+    attrs_.shape_proto = *shape_proto;
+    attrs_.shape = TensorShape(*shape_proto);
+    attrs_.name = node_def.name();
+
+    Tensor tensor(attrs_.dtype, attrs_.shape);
+    auto tensor_flat = tensor.flat<T>();
+    for (int64_t i = 0; i < tensor_flat.size(); i++) {
+      tensor_flat(i) = T(0.0f);
+    }
+
+    TRT_ShapedWeights weights;
+    TF_RETURN_IF_ERROR(
+        TfTensorToTrtWeights(tensor, params_->weight_store, &weights));
+
+    // Only push outputs during validation and when outputs are expected.
+    if (params_->validation_only && params_->outputs != nullptr) {
+      AddOutput(TRT_TensorOrWeights(weights));
+    }
+    return Status::OK();
+  }
+
+  Status Validate() {
+    const auto& node_def = params_->node_def;
+    if (params_->use_implicit_batch) {
+      return errors::Unimplemented("Implicit batch mode not supported, at ",
+                                   node_def.name());
+    }
+
+    StatusOr<DataType> dtype = GetAttrValue<DataType>("dtype");
+    TRT_ENSURE_OK(dtype);
+    attrs_.dtype = *dtype;
+
+    switch (attrs_.dtype) {
+      case DT_FLOAT:
+        return ValidateImpl<float>();
+      case DT_HALF:
+        return ValidateImpl<Eigen::half>();
+      default:
+        // Note: this should have been caught by ValidateNodeDefDataType, but
+        // the compiler expects that all paths be handled in switch.
+        return errors::Unimplemented("Data type ", DataTypeString(attrs_.dtype),
+                                     " is not supported for ", node_def.op(),
+                                     ", at ", node_def.name());
+    }
+  }
+
+  template <typename T>
+  Status ConvertImpl() {
```

Comment:
Could we have a common helper function for `ConvertVariableV2` and `ConvertReadVariableOp` to remove duplication while reading the variable values?